### PR TITLE
TASK: Complete the refactoring to using symfony/yaml

### PR DESCRIPTION
**Pull request by ComiR**
@radmiraal started this refactoring long ago. Now I finally got around "finishing" it.

As you can see by the `"`s, it is still a work in progress - so it's not final yet.
Everyone interested in using this package (f.e. with Neos 4) should probably have a look at this branch and try it with their config. I cannot guarantee it will work right now, i.e. I haven't tried it with any Active Directory service.

### CODE REVIEW

**Comment by ComiR** [comment]
I'm using this now since a while and it seems to be working just fine (even with AD ;)).

**Comment by Sebobo** [comment]
Awesome! As I finally was able to update my customers project to Neos 4.3 I can test & review this.

**Comment by ComiR** [comment]
It's even compatible with Flow 6 and Neos 5!

**Comment by Sebobo** [comment]
The upgrade seemed fine so far but we have trouble with the authentication. Currently we check whether it's the issue of the LDAP server or something else.

File: Classes/Security/Authentication/Provider/NeosBackendLdapProvider.php [hunk 1]
```diff
@@ -24,54 +25,53 @@
...
+        try {
+            $lastName = $this->eelEvaluator->evaluate($mapping['lastName'], $eelContext);
+        } catch (\Exception $exception) {
+            // todo: logging
             $lastName = 'none';
         }
 
-        $userService = new \Neos\Neos\Domain\Service\UserService();
-        $user = $userService->createUser(
+        $user = $this->userService->createUser(
```
Comment (Sebobo) [inline Classes/Security/Authentication/Provider/NeosBackendLdapProvider.php:67]: I think we should get rid of the usage of the `UserService`.

That's seems to be the only reason for the Neos dependency which makes the package unusable for Flow apps.

Also the `createUser` method will even create a personal workspace or might even assume the role `Neos.Neos:Editor` which is also not intended for frontend users.

I think we can replace that with a custom method in this class that could offer a way to still do those steps if needed in an app.

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1 @@
+*.yaml.example linguist-language=YAML
```
Comment (kdambekalns) [inline .gitattributes:1]: What does this do, and is it generally useful?

**Review by kdambekalns — COMMENTED** [review]
One question on the `.gitattributes` and then more generally speaking: Should this just be merged into the [task/symfony-yaml-refactor](https://github.com/neos/ldap/tree/task/symfony-yaml-refactor) branch so we can merge the current master into that to see what happens?

Time has passed and we have seen releases in between, so…

**Comment by kdambekalns** [comment]
I'll merge this and open a new PR based on the [neos:task/symfony-yaml-refactor](https://github.com/neos/ldap/tree/task/symfony-yaml-refactor) branch…

File: .gitattributes (hunk 2 above)
Comment (kdambekalns) [reply .gitattributes:1]: Seems to tell GitHub that those should be highlighted as YAML.

**Pull request merged**